WTRSimUnity/Assets/Scripts/SocketHandler.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using UnityEngine;
+using UnityEditor;
+using UnityEditor.PackageManager;
+using System.Collections;
+using UnityEngine.Assertions;
+using System.Threading;
+using System.Linq;
+using System.Text;
+
+//Singleton needed on game object
+
+// Credit to the University of Northampton, UK
+// http://www.eng.northampton.ac.uk/~espen/CSY2026/JavaServerCSClient.htm
+// CSY2026 Modern Networks
+
+/**
+ * Class that handles communication with Java
+ */
+[InitializeOnLoad]
+public class SocketHandler: MonoBehaviour
+{
+    int clientPort = 4513;
+    int serverPort = 4512;
```

Comment:
why these different?

Reply:
I should change the names "client" and "server" since it's not totally accurate in this context. If I used the same port for sending and receiving, I could wind up infinitely sending and receiving my own datagram.